test/src/test.c

```diff
@@ -232,3 +232,16 @@ p_test_fail(const char *prefix, const char *message) {
 	malloc_cprintf(NULL, NULL, "%s%s\n", prefix, message);
 	test_status = test_status_fail;
 }
+
+void
+strncpy_cond(void *dst, const char *src, bool cond) {
+	if (cond) {
```

Comment:
I can't see why this `if` branch is necessary.

Reply:
@yinan1048576 : This condition and the size n below are both workarounds for the false positives of the `stringop-overflow` check. The actual code will never be reached when cond == false. Since this is test code, I went extra conservative and included both.